src/__tests__/BrahmosES6class.test.js

```diff
@@ -0,0 +1,354 @@
+      constructor() {
+        super();
+        this.state = {};
+      }
+      // UNSAFE_componentWillMount() {
+      //   lifeCycles.push('will-mount');
+      // }
+      componentDidMount() {
+        lifeCycles.push('did-mount');
+      }
+      // UNSAFE_componentWillReceiveProps(nextProps) {
+      //   lifeCycles.push('receive-props', nextProps);
+      // }
+      shouldComponentUpdate(nextProps, nextState) {
+        lifeCycles.push('should-update', nextProps, nextState);
+        return true;
+      }
+      // UNSAFE_componentWillUpdate(nextProps, nextState) {
+      //   lifeCycles.push('will-update', nextProps, nextState);
+      // }
+      componentDidUpdate(prevProps, prevState) {
+        lifeCycles.push('did-update', prevProps, prevState);
+      }
+      componentWillUnmount() {
+        lifeCycles.push('will-unmount');
+      }
+      render() {
+        return <span className={this.props.value} />;
+      }
+    }
+    class Outer extends Brahmos.Component{
+      constructor(props){
+        super(props);
+        this.state={
+          isFooVisible: this.props.visible
+        }
+      }
+      unmountFoo(callback){
+        this.setState({
+          isFooVisible: false
+        },()=>callback)
+      }
+      render(){
+        if(this.state.isFooVisible){
+          return <Foo value={this.props.value}/>;
+        }
+        return <div/>;
+      }
+    }
+    test(<Outer visible value="foo" />, 'SPAN', 'foo');
+    expect(lifeCycles).toEqual(['did-mount']);
+    lifeCycles = []; // reset
+    const instance = test(<Outer visible value="bar" />, 'SPAN', 'bar');
+    expect(lifeCycles).toEqual([
+      'should-update', freeze({value: 'bar'}), {},
+      'did-update', freeze({value: 'foo'}), {},
+    ]);
+    lifeCycles = []; // reset
+    instance.unmountFoo(()=>expect(lifeCycles).toEqual(['will-unmount']));
+  });
+
+});
+
+
+/** Currently unsupported tests */
+
+// Brahmos hasn't installed prop-types
+
+// it('renders based on context in the constructor', () => {
+//   class Foo extends Brahmos.Component {
+//     constructor(props, context) {
+//       super(props, context);
+//       this.state = {tag: context.tag, className: this.context.className};
+//     }
+//     render() {
+//       const Tag = this.state.tag;
+//       return <Tag className={this.state.className} />;
+//     }
+//   }
+//   Foo.contextTypes = {
+//     tag: PropTypes.string,
+//     className: PropTypes.string,
+//   };
+
+//   class Outer extends Brahmos.Component {
+//     getChildContext() {
+//       return {tag: 'span', className: 'foo'};
+//     }
+//     render() {
+//       return <Foo />;
+//     }
+//   }
+//   Outer.childContextTypes = {
+//     tag: PropTypes.string,
+//     className: PropTypes.string,
+//   };
+//   test(<Outer />, 'SPAN', 'foo');
+// });
+
+
+// Don't have componentWillMount lifecycle method
+
+// it('renders only once when setting state in componentWillMount', () => {
```

Comment:
Same for this. We can delete it. We will not implement deprecated apis on Brahmos.